mypy/typeops.py

```diff
@@ -564,7 +564,8 @@ def callable_type(fdef: FuncItem, fallback: Instance,
     return CallableType(
         args,
         fdef.arg_kinds,
-        [None if argument_elide_name(n) else n for n in fdef.arg_names],
+        [None if argument_elide_name(n) or i < fdef.num_pos_only else n
+         for i, n in enumerate(fdef.arg_names)],
```

Comment:
Hmm could we instead set a value in `arg_names` to `None` to mark it as a positional-only argument? If this is possible, it seems cleaner that using `num_pos_only`.